tests/test_tools.py

```diff
@@ -27,5 +32,114 @@ def test_ClientRedirectServer(self):
         self.assertEqual(httpd.query_params.get('code'), code)
 
 
+class _Namespace(object):
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+
+class TestRunFlow(unittest2.TestCase):
+    @mock.patch.object(sys, 'argv',
+        [sys.executable, '--noauth_local_webserver'])
+    @mock.patch('oauth2client.tools.logging')
+    @mock.patch('oauth2client.tools.input')
+    def test_run_flow_no_webserver(self, input_mock, logging_mock):
+        flow = mock.Mock()
+        storage = mock.Mock()
+        credentials = mock.Mock()
+
+        input_mock.return_value = 'auth_code'
+        flow.step1_get_authorize_url.return_value = 'http://example.com/auth'
+        flow.step2_exchange.return_value = credentials
+
+        # Successful exchange.
+        returned_credentials = tools.run_flow(flow, storage)
+
+        self.assertEqual(credentials, returned_credentials)
+        self.assertEqual(flow.redirect_uri, OOB_CALLBACK_URN)
+        flow.step2_exchange.assert_called_once_with('auth_code', http=None)
+        storage.put.assert_called_once_with(credentials)
+        credentials.set_store.assert_called_once_with(storage)
+
+        # Same thing, but with explicit flags
+        flags = _Namespace(noauth_local_webserver=True, logging_level='INFO')
```

Comment:
Yeah I suppose. py2.6 will throw a fit, but I can skip the test as it none of this works on 2.6 anyway.